### ISSUE TITLE
Support for Merge operations on Snowflake queries

### ISSUE THREAD

**Issue by bhavesh91** [tokern/data-lineage#13]
We are having a requirement to parse the SQL queries that has MERGE operations - is there a plan to have a functionality to incorporate this ?

**Comment by vrajat** [comment]
Yes. I am assuming you are referring to the [MERGE statement](https://docs.snowflake.com/en/sql-reference/sql/merge.html) from Snowflake docs. Can you provide more info on how and why you are considering using data-lineage? This will help me prioritize the request. Can you fill in this [survey](https://forms.gle/p2oEQBJnpEguhrp3A)? 

Also will you be able to test an alpha build?

**Comment by sragavanvijay** [comment]
Hello @vrajat, Is the capability exists to connect to Snowflake directly and extract metadata to build lineage? Is there a plan to have this functionality built in the near future??

**Comment by vrajat** [comment]
Hi @sragavanvijay , data-lineage can pull the catalog from Snowflake. There is a query parser plugin to parse snowflake query history and build the lineage. A couple of examples that demo the current state:

* [Store and extract Snowflake Query History](https://tokern.io/blog/data-lineage-on-snowflake/)
* [Query Parser Example Notebook](https://tokern.io/docs/data-lineage/example)

So  the building blocks are there. Some scripting is required. I am working on packaging this better. Can you fill in this [survey](https://forms.gle/p2oEQBJnpEguhrp3A) so that I can better understand your requirements?
